src/db.js

```diff
@@ -29,9 +29,44 @@ class Database {
     }
 }
 
+class CentralLedgerDbError extends Error {
+    constructor(code, message) {
+        super(message);
+        this.code = code;
+    }
+}
+
+const ErrorCodes = Object.freeze({
+    PARTICIPANT_NOT_FOUND: 'PARTICIPANT_NOT_FOUND',
+});
+
+const Errors = {
+    participantNotFound: (fspId) => new CentralLedgerDbError(ErrorCodes.PARTICIPANT_NOT_FOUND, `${fspId} not found`),
+};
+
 class CentralLedgerDatabase extends Database {
-    constructor (config) {
-        super(config);
+    constructor (config, clientConstructor = Knex) {
+        super(config, clientConstructor);
+
+        /**
+         * Error types for this class. Note that we do not use symbols, or "instanceof" because if
+         * different versions of this code are used in different places, for example by transitive
+         * dependency, the symbols or instance types will not be the same.
+         */
+        this.ErrorCodes = ErrorCodes;
+    }
+
+    /**
+     * Verifies an error type
+     *
+     * Note that we do not use symbols, or "instanceof" because if different versions of this code
+     * are used in different places, for example by transitive dependency, the symbols or instance
+     * types will not be the same.
+     *
+     * @returns boolean - error is of the type supplied
+     */
+    errorIs(err, code) {
+        return code in this.ErrorCodes && err.code === code;
```

Comment:
Is this more resilient than `this.ErrorCodes[code]`?

Reply:
IIRC it's equivalent to `hasOwnProperty` rather than `[]`

Reply:
I would rather name that function `validateError` or something that starts with any other verb, most likely `is`, because the `errorIs` makes me thing that it will return something, like `error is blah`.

Reply:
You mean the function name makes you think it would return some code or something that indicated the error type, rather than a boolean value indicating that the error was of the supplied type?

Reply:
I made a change, assuming I correctly interpreted your objection (which seems reasonable).

Reply:
Yes this is what I meant :)